crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -546,6 +546,41 @@ impl Mesh {
         self
     }
 
+    /// Inverts the winding of the indices such that all counter-clockwise triangles are now
+    /// clockwise and vice versa.
+    ///
+    /// Does nothing if no [`Indices`] are set.
+    pub fn invert_winding(&mut self) -> Result<(), MeshWindingInvertError> {
+        if self.primitive_topology != PrimitiveTopology::TriangleList {
+            return Err(MeshWindingInvertError::WrongTopology);
+        }
+
+        fn invert<'a, I: 'static>(
+            indices: impl Iterator<Item = &'a mut [I]>,
+        ) -> Result<(), MeshWindingInvertError> {
+            for chunk in indices {
+                let [_, b, c] = chunk else {
+                    return Err(MeshWindingInvertError::AbruptIndicesEnd);
```

Comment:
We should do an early check on the number of indices to avoid stopping half-way through mutating a mesh, especially since this is a relatively expensive operation.

Reply:
The problem is that rusts chunk iterator is a slice. I'm not sure how well rust does optimization but I think we would need to use unsafe here when doing a check at the top. Do we want that?

Reply:
Right, that's fair. Okay, IMO just add a warning in the docs for now then.

Reply:
I feel like it might still be desirable to do a quick check that `indices.len()%3 == 0` at the start of `fn invert`, just for the sake of not leaving the mesh partially flipped, and returning early in case of an error.
We can have `invert` accept an `&mut [I]` instead of an `impl Iterator<Item = &'a mut [I]>`